### ISSUE TITLE
train.py: do_resume=True should reject structural config drift from the resumed-from run

### ISSUE THREAD

**Issue by mmcdermott** [payalchandak/EveryQuery#91]
## What I found

`pretrain/train.py::main` (currently `src/every_query/train.py` on `dev` before #90 lands) supports `do_resume=True` but does **no** check that the on-disk config.yaml from the original run matches the config the user is now passing. The flow on resume:

1. `ckpt_path = find_checkpoint_path(output_dir)` — picks a checkpoint file.
2. `output_dir/config.yaml` is left untouched (good — preserves the original run's config).
3. `cfg` (the *new*, potentially drifted, OmegaConf) is used to instantiate the datamodule, lightning module, and trainer.
4. `trainer.fit(ckpt_path=ckpt_path)` resumes from the checkpoint.

There's no cross-check between `cfg` and `OmegaConf.load(output_dir / "config.yaml")`.

## Why this matters

- **Hard breakage** on architectural drift (`hidden_size`, `num_layers`, `num_attention_heads`, vocab size): `load_from_checkpoint` will crash at weight load. That's loud and visible.
- **Silent drift** on non-architectural knobs: `max_seq_len`, `batch_size`, `seq_sampling_strategy`, `static_inclusion_mode`, `query.codes`, `seed`. A resume with a different `max_seq_len` would produce a run that looks like the original on paper but isn't — runs posted to wandb / GCS are no longer comparable.
- **Partial drift** on training-schedule params (`max_steps`, `val_check_interval`, early-stopping patience): arguably legitimate things to override on resume. These shouldn't raise.

So the right behavior is: **fail loudly on drift in everything except an explicit deny-list of "fine to differ" keys.**

## Prior art in MEDS-EIC-AR

[`MEDS_EIC_AR/training/files.py`](https://github.com/mmcdermott/MEDS_EIC_AR/blob/main/src/MEDS_EIC_AR/training/files.py) already solves this. Three pieces, directly portable:

1. **[`ALLOWED_DIFFERENCE_KEYS`](https://github.com/mmcdermott/MEDS_EIC_AR/blob/main/src/MEDS_EIC_AR/training/files.py#L7-L16)** — deny-list of keys that legitimately differ on resume (the current EQ equivalent would add `trainer.max_steps` and friends):
   ```python
   ALLOWED_DIFFERENCE_KEYS = {
       "do_resume",
       "output_dir",
       "do_overwrite",
       "datamodule.num_workers",
       "trainer.logger.save_dir",
       "trainer.callbacks.model_checkpoint.dirpath",
       "trainer.default_root_dir",
       "log_dir",
   }
   ```

2. **[`diff_configs(new, old)`](https://github.com/mmcdermott/MEDS_EIC_AR/blob/main/src/MEDS_EIC_AR/training/files.py#L63-L140)** — recursive deep-diff returning a `{dotted_key: human_message}` dict. Handles enum parameters (`seq_sampling_strategy`, `padding_side`, `static_inclusion_mode`) with case-insensitive comparison via `resolve_enum_value`, so `"RANDOM"` vs `SubsequenceSamplingStrategy.RANDOM` is not flagged as a diff.

3. **[`validate_resume_directory(output_dir, cfg)`](https://github.com/mmcdermott/MEDS_EIC_AR/blob/main/src/MEDS_EIC_AR/training/files.py#L143-L223)** — loads on-disk `config.yaml`, runs `diff_configs`, strips anything in `ALLOWED_DIFFERENCE_KEYS`, raises `ValueError` with a formatted per-key message if anything's left.

This is strictly better than my earlier pseudocode — deep recursive diff catches drift at any nesting level, and deny-list semantics default to strict (new fields a future PR adds are automatically guarded).

## Proposed port

Copy `diff_configs` + `validate_resume_directory` into (post-#90) `src/every_query/pretrain/` as `pretrain/resume_check.py` (or similar). Tweaks for EQ:

- `ALLOWED_DIFFERENCE_KEYS` adds the training-schedule overrides we explicitly want to allow on resume:
  ```python
  ALLOWED_DIFFERENCE_KEYS = {
      "do_resume", "do_overwrite", "output_dir",
      "trainer.max_steps",            # legit to bump on resume
      "trainer.max_epochs",           # same
      "trainer.callbacks.early_stopping.patience",
      "datamodule.num_workers", "datamodule.pin_memory",
      "trainer.logger.save_dir",
      "trainer.callbacks.model_checkpoint.dirpath",
      "trainer.default_root_dir",
      "log_dir",
  }
  ```
- `STR_ENUM_PARAMS` stays identical — EQ uses the same three MTD enums.
- Call site in `train.main`:
  ```python
  elif cfg.do_resume:
      logger.info(f"Resuming training in existing output directory {output_dir}.")
      validate_resume_directory(output_dir, cfg)  # raises on structural drift
      ckpt_path = find_checkpoint_path(output_dir)
  ```

## Acceptance criteria

- [ ] `pretrain/resume_check.py` (or equivalent) landed with `diff_configs`, `validate_resume_directory`, and an `ALLOWED_DIFFERENCE_KEYS` tuned for EQ.
- [ ] Call inserted in `train.main`'s `do_resume` branch, *before* `find_checkpoint_path`.
- [ ] Regression tests:
  - resume after bumping `trainer.max_steps` — passes (it's in `ALLOWED_DIFFERENCE_KEYS`).
  - resume after changing `datamodule.config.max_seq_len` — raises with a readable diff.
  - resume with `seq_sampling_strategy: "RANDOM"` vs `SubsequenceSamplingStrategy.RANDOM` — does NOT raise (enum handling).

Found while discussing #31 / PR #86.

**Comment by mmcdermott** [comment]
Closing — addressed by PR #94, merged to `dev`.
